uktrade/data-hub-frontend
# Reactify Large Capital Profile details section

## Description of change

I am currently working on reactifying the company Large Capital Profile tab. As this is a very large piece of work, in order to reduce the review burdern I will split this task into four PRs and merge each of these into a `feature-merge` branch.

This is the first PR, which contains the following changes:
- A skeleton page in React to contain the various tables/forms, and set up the logic to display them
- Reactifying the profile details table
- Refactoring the existing React form to use data collected via Redux
- Moving all the modified code into the `modules` directory

Once this has been merged, I will raise two subsequent PRs to reactify the Requirements and Location sections followed by a final cleanup PR (which will contain updated tests and removal of all legacy/dead code).

For convenience [this commit](https://github.com/uktrade/data-hub-frontend/pull/5480/commits/a866e3cdffe7a8bdf5ed85613fcfc7810f4f8e55) only contains existing code that has been moved.

## Test instructions

Navigate to a company with a large capital profile. You should be able to view and edit the details. There are also several bugfixes:
- It is now possible to submit the form with the two currency fields blank (this would previously cause the task to fail)
- Any changes to the description will now save in the correct place, meaning that it is possible to 'complete' the details section (this was not possible previously as the description was wrongly being saved in the location tab 🤷‍♂️)

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'
